Dockerfile

```diff
@@ -1,76 +1,86 @@
-RUN mkdir -p /opt/i3-data/photon-tables/splines/
-RUN ls /opt/i3-data
-RUN wget -nv -t 5 -O /opt/i3-data/photon-tables/splines/InfBareMu_mie_abs_z20a10_V2.fits \
-        http://prod-exe.icecube.wisc.edu/spline-tables/InfBareMu_mie_abs_z20a10_V2.fits
-RUN wget -nv -t 5 -O /opt/i3-data/photon-tables/splines/InfBareMu_mie_prob_z20a10_V2.fits \
-        http://prod-exe.icecube.wisc.edu/spline-tables/InfBareMu_mie_prob_z20a10_V2.fits
-RUN wget -nv -t 5 -O /opt/i3-data/photon-tables/splines/cascade_single_spice_bfr-v2_flat_z20_a5.abs.fits \
-        http://prod-exe.icecube.wisc.edu/spline-tables/cascade_single_spice_bfr-v2_flat_z20_a5.abs.fits
-RUN wget -nv -t 5 -O /opt/i3-data/photon-tables/splines/cascade_single_spice_bfr-v2_flat_z20_a5.prob.fits \
-        http://prod-exe.icecube.wisc.edu/spline-tables/cascade_single_spice_bfr-v2_flat_z20_a5.prob.fits
-RUN wget -nv -t 5 -O /opt/i3-data/photon-tables/splines/cascade_effectivedistance_spice_bfr-v2_z20.eff.fits \
-        http://prod-exe.icecube.wisc.edu/spline-tables/cascade_effectivedistance_spice_bfr-v2_z20.eff.fits
-
-# install baseline GCDs
-RUN mkdir /opt/i3-data/baseline_gcds
-RUN wget -nv -N -t 5 -P /opt/i3-data/baseline_gcds -r -l 1 -A *.i3* -nd https://icecube:skua@convey.icecube.wisc.edu/data/user/followup/baseline_gcds/ && \
-    chmod -R u+rwX,go+rX,go-w /opt/i3-data/baseline_gcds
-RUN ls /opt/i3-data/baseline_gcds
+#
+# Manually compile Pulsar
+#
+WORKDIR /local
+RUN git clone https://github.com/apache/pulsar-client-cpp
+RUN cd pulsar-client-cpp && cmake -DBUILD_TESTS=OFF -DCMAKE_BUILD_TYPE=Release . && make -j2 && make install
+RUN git clone https://github.com/apache/pulsar-client-python
+RUN cd pulsar-client-python && git submodule update --init \
+    && cmake -B build \
+    && cmake --build build && cmake --install build \
+    && python3 ./setup.py bdist_wheel
 
 
 #
-# Get IceTray Setup
+# Now make the prod image
 #
-
-# add realtime_gfu python checkout from V21-06-00
-RUN svn co http://code.icecube.wisc.edu/svn/meta-projects/realtime/releases/V21-06-00/realtime_gfu \
-        /usr/local/icetray/realtime_gfu --username=icecube --password=skua --no-auth-cache && \
-    ln -sf /usr/local/icetray/realtime_gfu/python /usr/local/icetray/lib/icecube/realtime_gfu
-# add realtime_hese
-RUN svn co http://code.icecube.wisc.edu/svn/meta-projects/realtime/releases/V21-06-00/realtime_hese \
-        /usr/local/icetray/realtime_hese --username=icecube --password=skua --no-auth-cache && \
-    ln -sf /usr/local/icetray/realtime_hese/python /usr/local/icetray/lib/icecube/realtime_hese
-# add realtime_tools
-RUN svn co http://code.icecube.wisc.edu/svn/meta-projects/realtime/releases/V21-06-00/realtime_tools \
-        /usr/local/icetray/realtime_tools --username=icecube --password=skua --no-auth-cache && \
-    ln -sf /usr/local/icetray/realtime_tools/python /usr/local/icetray/lib/icecube/realtime_tools
+FROM icecube/icetray:icetray-prod-v1.4.1-ubuntu22.04-2022-11-29 as prod 
```

Comment:
Does this tag need to be the same as the first `FROM`, as in `*v1.4.1-ubuntu22.04-2022-11-29`? If so, this may benefit by being a variable, so we don't have a version/tag mismatch in the future.

Reply:
Not necessarily.  It's probably a good idea to keep them the same though.  I'm unsure how you would make this a variable in a Dockerfile.

Reply:
I think it's similar to shell syntax `$myvar`: https://docs.docker.com/engine/reference/builder/#arg